src/py42/services/_connection.py

```diff
@@ -275,6 +275,7 @@ def _handle_error(method, url, response):
         msg = u"No response was returned for {} request to {}.".format(method, url)
         raise Py42Error(msg)
 
+    debug.logger.debug(u"Response\n", str(response))
```

Comment:
would this suffice?!!

Reply:
Looking at the ticket, it says we want to log every time debug is turned on.
This logging should happen outside of the `_handle_error()` method

Reply:
Line 192/193 area would work

Reply:
Do you get any output from this though? When I do

```
            debug.logger.debug("TEST")
            debug.logger.debug(u"Response\n", str(response))
            debug.logger.debug("TEST")
```

The output from that part is just :

```
TEST
TEST
```

Reply:
it did work for me.

Reply:
"_it says we want to log every time debug is turned on_"
As per logging module implementation, we need not check log level explicitly, for all level "debug and above" all debug messages would be displayed.

Reply:
I just missed the return statement for success cases and was under the impression that `handle_error` is called always irrespective of whether there is an error or not, and does nothing if there is no error.

Changed now.

Reply:
I still am not seeing any Response logging.
I see the requests getting logged, as they did before, but the specific line you added is not showing up for me

Reply:
I think it is because you are passing multiple args to the `debug` method. You only need to pass it a string (1 arg).

Also, the __str__ of a response is just `<Response [200]>`. It does not contain any of the response data. We want to include both the status code _and_ the response data.